python/dataset_plotting.py

```diff
@@ -0,0 +1,239 @@
+# Dataset imports
+from dataset import Data, dataset
+import numpy as np
+# Plotly imports
+from plotly.offline import init_notebook_mode, iplot
+init_notebook_mode(connected=True)
+import plotly.graph_objs as go
+
+#===============================================================================
+
+# Plot a 2D slice through a 3D dataset with a slider to adjust the position of
+# the slice in the third dimension
+def plot_sliceviewer(input_data, field=None):
+
+    # Delay import to here, as ipywidgets is not part of the base plotly package
+    try:
+        from ipywidgets import interactive, HBox, VBox
+    except ImportError:
+        print("Sorry, the sliceviewer requires ipywidgets which was not found on this system")
+        return
+
+    dims = input_data.dimensions()
+    if (len(dims) > 2) and (len(dims) < 5):
+
+        # Get spatial extents by iterating over the coordinates, and also search
+        # for data fields
+        axmin = []
+        axmax = []
+        axlab = []
+        fields = []
+        nx = []
+        ifield = 0
+        for var in input_data:
+            if var.is_coord:
+                values = var.numpy
+                axmin.append(values[0] - 0.5*(values[1]-values[0]))
+                axmax.append(values[-1] + 0.5*(values[-1]-values[-2]))
+                axlab.append([var.name, var.unit])
+            elif var.is_data:
+                fields.append([var.name, var.unit])
+                if var.name == field:
+                    ifield = len(fields) - 1
+
+        ratio = (axmax[1] - axmin[1]) / (axmax[0] - axmin[0])
+
+        if (field is None) and (len(fields) > 1):
+            raise RuntimeError("More than one data field found! Please specify which one to display")
```

Comment:
Can you specify how, otherwise this is not clear: For example, `Use, e.g., plot(dataset.subset(Data.Value, "sample")) to plot data for...`?